crates/accelerate/src/gate_direction.rs

```diff
@@ -142,8 +157,505 @@ where
+                            packed_inst.op.name()
+                        )));
+                    }
+                }
+                _ => {}
+            }
+        }
+        // No matching replacement found
+        if gate_complies(packed_inst, &[op_args1, op_args0])
+            || has_calibration_for_op_node(py, dag, packed_inst, &[op_args1, op_args0])?
+        {
+            return Err(TranspilerError::new_err(format!("{} would be supported on {:?} if the direction was swapped, but no rules are known to do that. {:?} can be automatically flipped.", packed_inst.op.name(), op_args, vec!["cx", "cz", "ecr", "swap", "rzx", "rxx", "ryy", "rzz"])));
+            // NOTE: Make sure to update the list of the supported gates if adding more replacements
+        } else {
+            return Err(TranspilerError::new_err(format!(
+                "{} with parameters {:?} is not supported on qubits {:?} in either direction.",
+                packed_inst.op.name(),
+                packed_inst.params_view(),
+                op_args
+            )));
+        }
+    }
+
+    for (node, op_blocks) in ops_to_replace {
+        let NodeType::Operation(packed_inst) = &dag.dag()[node] else {
+            panic!("PackedInstruction is expected");
+        };
+        let OperationRef::Instruction(py_inst) = packed_inst.op.view() else {
+            panic!("PyInstruction is expected");
+        };
+        let new_op = py_inst
+            .instruction
+            .bind(py)
+            .call_method1("replace_blocks", (op_blocks,))?;
+
+        dag.py_substitute_node(dag.get_node(py, node)?.bind(py), &new_op, false, false)?;
+    }
+
+    for (node, replacemanet_dag) in nodes_to_replace {
+        dag.py_substitute_node_with_dag(
+            py,
+            dag.get_node(py, node)?.bind(py),
+            &replacemanet_dag,
+            None,
+            true,
+        )?;
+    }
+
+    Ok(dag)
+}
+
+// Check whether the dag as calibration for a DAGOpNode
+fn has_calibration_for_op_node(
+    py: Python,
+    dag: &DAGCircuit,
+    packed_inst: &PackedInstruction,
+    qargs: &[Qubit],
+) -> PyResult<bool> {
+    let py_args = PyTuple::new_bound(py, dag.qubits().map_indices(qargs));
+
+    let dag_op_node = Py::new(
+        py,
+        (
+            DAGOpNode {
+                instruction: CircuitInstruction {
+                    operation: packed_inst.op.clone(),
+                    qubits: py_args.unbind(),
+                    clbits: PyTuple::empty_bound(py).unbind(),
+                    params: packed_inst.params_view().iter().cloned().collect(),
+                    extra_attrs: packed_inst.extra_attrs.clone(),
+                    #[cfg(feature = "cache_pygates")]
+                    py_op: packed_inst.py_op.clone(),
+                },
+                sort_key: "".into_py(py),
+            },
+            DAGNode { node: None },
+        ),
+    )?;
+
+    dag.has_calibration_for(py, dag_op_node.borrow(py))
+}
```

Comment:
For this function, is there a reason why `has_calibration_for_index` can't be used? From what I can remember `has_calibration_for` works for `DAGOpNodes` that are already valid within the `DAGCircuit`, so a `NodeIndex` should work just as fine.

Reply:
`has_calibration_for_index` uses the qubits order of the node whose `NodeIndex` is given. This is checked already in line 195. However, here we want to check if there is a calibration def for the same node type and params, but with a reversed qubits order, a combination which might not exist necessarily in the dag. We could theoretically extend the `has_calibration_for_index` function with an Option arg to indicate a custom qubits order to check on. I'm not sure it's worthwhile since the path is being deprecated anyway. What do you say?

Reply:
That's fine then. I just wanted to understand the reasoning behind this, but what you said makes sense. If anything we can introduce another way to check for calibrations in a follow-up.